core/src/error.rs

```diff
@@ -36,6 +36,26 @@ pub struct InvalidMultihash(pub Vec<u8>);
 #[cfg_attr(feature = "std", derive(Error), error("Failed to retrieve block {0}."))]
 pub struct BlockNotFound(pub Cid);
 
+/// Error during Serde operations.
+#[cfg(feature = "serde-codec")]
+#[derive(Clone, Debug)]
+#[cfg_attr(feature = "std", derive(Error), error("Serde error: {0}."))]
+pub struct SerdeError(String);
+
+#[cfg(feature = "serde-codec")]
+impl serde::de::Error for SerdeError {
+    fn custom<T: std::fmt::Display>(msg: T) -> Self {
```

Comment:
Currently building `cargo build --no-default-features --features serde-codec --target thumbv6m-none-eabi --manifest-path core/Cargo.toml` fails due to Serde. So that might be difficult to test.

Reply:
Hm. Just building with `#![no_std]` should catch this, even without `thumbv6m-none-eabi`.